boards/slstk3401a/board.c

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2015-2018 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     boards_slstk3401a
+ * @{
+ *
+ * @file
+ * @brief       Board specific implementations SLSTK3401A board
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Bas Stottelaar <basstottelaar@gmail.com>
+ *
+ * @}
+ */
+
+#include "board.h"
+#include "board_common.h"
+
+#include "periph/gpio.h"
+
+void board_init(void)
+{
+    /* initialize the CPU */
+    cpu_init();
+
+    /* perform common board initialization */
+    board_common_init();
+
+#ifdef MODULE_SI70XX
```

Comment:
is it possible to not use this module physically on the board? I mean, that you can disable it somehow so you need this guard?

Reply:
Well, removing this condition will always enable the sensor, even if you don't use it. I would therefore not want to remove it.

Reply:
So what you mean is that this pin can turn on/off the sensor? If that's the case, then this is a kind of "software detachment".

IMHO it looks a bit strange to have it here, but I don't have in mind right now a better way to integrate this functionality.

Reply:
Just to clarify: this pin is not directly connected to the enable line of the Si7021: it's connected to a board-specific IC that will toggle VDD for the Si7021.

So to use this Si7021, you have to enable this pin to supply it VDD. The reason to disable it, is to not take into account its consumption when profiling it.

Reply:
@basilfx Yes is what I thought, a real physical software controlled switch. That's why I didn't have an idea on how to integrate it, but I guess we should leave it as it is here and if the case comes across again we find a better integration.